### PR TITLE
Implemented name mapping for types and unit tests

### PR DESCRIPTION
**Pull request by RachelTucker**

### CODE REVIEW

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.spectralogic.ds3autogen.api.models.Ds3NameMapper;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class Ds3NameMapperParserImpl implements Ds3NameMapperParser {
+
+    private final ObjectMapper objectMapper;
+
+    public Ds3NameMapperParserImpl() throws IOException {
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java]: What throws IOException?

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        return builder.build();
+    }
+
+    protected static ImmutableList<Ds3EnumConstant> convertAllEnumConstants(
+            final ImmutableList<Ds3EnumConstant> enumConstants,
+            final NameMapper nameMapper) {
+        if (enumConstants == null || enumConstants.isEmpty()) {
+            return enumConstants;
+        }
+        ImmutableList.Builder<Ds3EnumConstant> builder = ImmutableList.builder();
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java]: Missing final

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        return builder.build();
+    }
+
+    protected static ImmutableList<Ds3Property> convertAllProperties(
+            final ImmutableList<Ds3Property> properties,
+            final NameMapper nameMapper) {
+        if (properties == null || properties.isEmpty()) {
+            return properties;
+        }
+        ImmutableList.Builder<Ds3Property> builder = ImmutableList.builder();
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java]: Missing final

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        return builder.build();
+    }
+
+    protected static ImmutableList<Ds3Element> convertAllElements(
+            final ImmutableList<Ds3Element> elements,
+            final NameMapper nameMapper) {
+        if (elements == null || elements.isEmpty()) {
+            return elements;
+        }
+        ImmutableList.Builder<Ds3Element> builder = ImmutableList.builder();
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java]: Missing final

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        return builder.build();
+    }
+
+    protected static ImmutableList<Ds3Annotation> convertAllAnnotations(
+            final ImmutableList<Ds3Annotation> annotations,
+            final NameMapper nameMapper) {
+        if (annotations == null || annotations.isEmpty()) {
+            return annotations;
+        }
+        ImmutableList.Builder<Ds3Annotation> builder = ImmutableList.builder();
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java]: Missing final

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        return builder.build();
+    }
+
+    protected static ImmutableList<Ds3AnnotationElement> convertAllAnnotationElements(
+            final ImmutableList<Ds3AnnotationElement> annotationElements,
+            final NameMapper nameMapper) {
+        if (annotationElements == null || annotationElements.isEmpty()) {
+            return annotationElements;
+        }
+        ImmutableList.Builder<Ds3AnnotationElement> builder = ImmutableList.builder();
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java]: Missing final

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    private static ObjectMapper initDs3NameMapper() {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new GuavaModule());
+        return objectMapper;
+    }
+
+    @Override
+    public Ds3NameMapper getMap(final InputStream stream) throws IOException {
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java]: Can this be made static?

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3autogen;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.spectralogic.ds3autogen.api.models.*;
+
+public class Ds3SpecConverter {
```
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java]: Why is the class public when all the methods appear to be protected?

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java (hunk 1 above)
Comment (RachelTucker) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java]: On line 44, the following throws the IOException:
     ByteStreams.toByteArray(stream)

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java (hunk 7 above)
Comment (RachelTucker) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java]: If its made static, then I  need to create a body for this function in its interface, which defeats the purpose of an interface.  If I do that, then yes, I think I can make it static.

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java (hunk 8 above)
Comment (RachelTucker) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3SpecConverter.java]: Missed it when I refactored the scope of the class.  I'll make it protected.

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java (hunk 7 above)
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java]: Oh, I see.  I didn't realize that this was an implementation.  No change is needed here then.

File: ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java (hunk 1 above)
Comment (rpmoore) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java]: Sure, but the only function called in the constructor does not throw an IOException.

Comment (RachelTucker) [inline ds3-autogen/src/main/java/com/spectralogic/ds3autogen/Ds3NameMapperParserImpl.java]: Oops.  Another missed item from code refactoring.  I'll remove it.

**Pull request merged**